bold.py

```diff
@@ -6,4 +7,50 @@ class BoldBlock(Block):
     '''
     def _render(self):
         return '<strong>{}</strong>'.format(self.content())
-      
\ No newline at end of file
+
+    @staticmethod
+    def parse(content):
+        '''
+        Input: a text content
+        Output: a tuple (int, int, BoldBlock)
+            [0]: start index,
+            [1]: end index,
+            [2]: created class,
+
+        Note, the index of start and end follow the convention of [start, end)
+        '''
+        pattern = r'([*]+)(.*?)([*]+)'
+        matches = re.search(pattern, content)
+        if matches:
+            head_asterisks_num = len(matches.group(1))
+            tail_asterisks_num = len(matches.group(3))
+            start_index = matches.start(1)
+            end_index = matches.end(3)
+
+            # follow the trigger convention of Github
+            # as long as head or tail has three asterisks, it's bold and italic
+            if (head_asterisks_num == 3 or tail_asterisks_num == 3) and \
+                head_asterisks_num+tail_asterisks_num >= 6:
+                # match Bold and Italic first
+                # ***test*** -> *test* -> pass to ItatlicBlock()
+                head_remain = max(head_asterisks_num - 2, 0)
+                tail_remain = max(tail_asterisks_num - 2, 0)
+                bold_content = '*' * head_remain + matches.group(2) + '*' * tail_remain
+                return (
+                    start_index,
+                    end_index,
+                    BoldBlock(content=bold_content)
+                )
+            elif head_asterisks_num >= 2 and tail_asterisks_num >= 2:
+                # match bold without italic
+                diff = head_asterisks_num - tail_asterisks_num
+                if diff >= 0:
+                    bold_content = '*' * diff + matches.group(2)
+                else:
+                    bold_content = matches.group(2) + '*' * -diff
+                return (
+                    start_index,
+                    end_index,
+                    BoldBlock(content=bold_content)
+                )
```

Comment:
I don't think those 2 branches makes sense. you can simply merge them. Beside, you don't need to worry the bold+italic case, let the nested parser to handle them.

Reply:
> I don't think those 2 branches makes sense. you can simply merge them. 

which part you think it doesn't make sense?

> Beside, you don't need to worry the bold+italic case, let the nested parser to handle them.

Yes, this is what i have now. in the bold and italic, `[*](3)abc[*](3)` will become `[*](1)abc[*](1)`, then it is an italic, so that the nested parser will handle it.

Reply:
I agree with Andy. Your solution is correct but it can be simplified and no need to consider bold+italic case. Such as `***a*****`, it will fail into the first if condition. The bold content can be `*a***`. The italic block can continue to parse it.

Reply:
I am not sure what's the logic that you guys think those two branches can be merged. 

What's your expectation for `****a*****` and `**a***`? 

This is the GitHub parser result :  
`****a*****` will become ****a*****
`**a***` will become **a***

you can see that they are the same. 

The convention that GitHub using is, only if either head or tail has exactly 3 asterisks and (head+tail)>=6 asterisks, it would become Bold and Italic, which means the bold content should have asterisks as **BOTH** prefix **and** postfix. Otherwise, parser will compare the diff of head and tail, and add the remaining asterisks as **EITHER** prefix **OR** postfix. I've also tried different Markdown parsers, they are showing the consistent result with Github. 

I personally feel like those two branches are different logic and not sure how to merge then together.

If we don't make Bold and Italic rule as a special case, taking Ying's example, `***a*****` falls into second if condition of my code, then it becomes `a**` which is not a bold and italic.